### PR TITLE
ci: make changelog-checker only validate PR number against main base

### PR DESCRIPTION
**Pull request by mikemorris**
Intent of this change is to avoid spurious changelog-checker failures on manual backport PRs, where cherry-picking the changelog entry from the original commit will intentionally not match the PR number of the backport (matching the original change instead).

My bash might need some revisions to make sure the logic is correct and syntax is valid.

### CODE REVIEW

**Comment by mikemorris** [comment]
@alvin-huang Any idea why this check is now just failing with exit code `1` instead of failing and commenting on the PR? I didn't change much, but is there a different way these should be written now (like with https://github.com/actions/github-script#comment-on-an-issue?) or is this an effect of editing this file in the PR?

File: .github/workflows/changelog-checker.yml [hunk 1]
```diff
@@ -23,13 +23,23 @@ jobs:
           fetch-depth: 0 # by default the checkout action doesn't checkout all branches
       - name: Check for changelog entry in diff
         run: |
+          pull_request_base_main=$(expr github.event.pull_request.base.ref = "main")
```
Comment (alvin-huang) [inline .github/workflows/changelog-checker.yml:26]: Try interpolating this with the github syntax `${{ github.event.pull_request.base.ref }}`. I don't think it works as written (I haven't tested)

Comment (mikemorris) [reply .github/workflows/changelog-checker.yml:26]:
```suggestion
          pull_request_base_main=$(expr ${{ github.event.pull_request.base.ref }} = "main")
```

File: .github/workflows/changelog-checker.yml [hunk 2]
```diff
@@ -23,13 +23,23 @@ jobs:
...
+            changelog_file_path=".changelog/*.txt"
+          fi
+
+          changelog_files=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/main")" -- ${changelog_file_path})
 
           # If we do not find a file in .changelog/, we fail the check
           if [ -z "$changelog_files" ]; then
             # Fail status check when no .changelog entry was found on the PR
-            echo "Did not find a .changelog entry and the 'pr/no-changelog' label was not applied. Reference - https://github.com/hashicorp/consul/pull/8387"
+            echo "Did not find a .changelog entry ${enforce_matching_pull_request_number}and the 'pr/no-changelog' label was not applied. Reference - https://github.com/hashicorp/consul/pull/8387"
```
Comment (kisunji) [inline .github/workflows/changelog-checker.yml:42]:
```suggestion
            echo "Did not find a .changelog entry ${enforce_matching_pull_request_number} and the 'pr/no-changelog' label was not applied. Reference - https://github.com/hashicorp/consul/pull/8387"
```

**Review by kisunji — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/428417.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 2a98ed5b7d625025ca6f099c725edf05c2c8d9ed onto `release/1.10.x` succeeded!

**Pull request merged**